### PR TITLE
[SPARK-23140][SQL] Add DataSourceV2Strategy to Hive Session state's planner

### PR DESCRIPTION
**Pull request by jerryshao**
## What changes were proposed in this pull request?

`DataSourceV2Strategy` is missing in `HiveSessionStateBuilder`'s planner, which will throw exception as described in [SPARK-23140](https://issues.apache.org/jira/browse/SPARK-23140).

## How was this patch tested?

Manual test.

### CODE REVIEW

**Comment by jerryshao** [comment]
@cloud-fan , please help to review, thanks!

File: sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveSessionStateBuilder.scala [hunk 1]
```diff
@@ -101,6 +102,7 @@ class HiveSessionStateBuilder(session: SparkSession, parentState: Option[Session
       override def strategies: Seq[Strategy] = {
         experimentalMethods.extraStrategies ++
           extraPlanningStrategies ++ Seq(
```
Comment (cloud-fan) [inline sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveSessionStateBuilder.scala]: can we change it to `super. strategies ++ Seq(HiveTableScans, Scripts)`?

**Comment by cloud-fan** [comment]
thanks, good catch!

File: sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveSessionStateBuilder.scala (hunk 1 above)
Comment (jerryshao) [reply sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveSessionStateBuilder.scala]: Sure, I will change it.

**Comment by cloud-fan** [comment]
LGTM

File: sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveSessionStateBuilder.scala (hunk 1 above)
Comment (jerryshao) [reply sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveSessionStateBuilder.scala]: Looks like the ordering matters, If I put Hive related strategies in the end, some unit tests will be failed.

File: sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveSessionStateBuilder.scala [hunk 2]
```diff
@@ -98,20 +98,7 @@ class HiveSessionStateBuilder(session: SparkSession, parentState: Option[Session
...
-          SpecialLimits,
-          InMemoryScans,
-          HiveTableScans,
-          Scripts,
-          Aggregation,
-          JoinSelection,
-          BasicOperators
-        )
-      }
+      override def strategies: Seq[Strategy] = Seq(HiveTableScans, Scripts) ++ super.strategies
```
Comment (cloud-fan) [inline sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveSessionStateBuilder.scala]: This breaks the assumption that `experimentalMethods.extraStrategies` should always run first.

I think we can just do:
```
override def extraPlanningStrategies: Seq[Strategy] =
  super.extraPlanningStrategies ++ customPlanningStrategies ++ Seq(HiveTableScans, Scripts)
```

Comment (jerryshao) [reply sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveSessionStateBuilder.scala]: OK, let me update it.

**Comment by cloud-fan** [comment]
thanks, merging to master/2.3!

**Pull request closed without merging**